snapcraft/storeapi/errors.py

```diff
@@ -334,6 +334,42 @@ def __init__(self, snap_name, response):
                          **response_json)
 
 
+class StoreMetadataError(StoreError):
+
+    __FMT_NOT_REGISTERED = (
+        'Sorry, try `snapcraft register {snap_name}` before trying to '
+        'update metadata.')
+
+    fmt = 'Received {status_code!r}: {text!r}'
+
+    def __init__(self, snap_name, response, metadata):
+        try:
+            response_json = response.json()
+        except (AttributeError, JSONDecodeError):
+            response_json = {}
+
+        if response.status_code == 404:
+            self.fmt = self.__FMT_NOT_REGISTERED
+        elif response.status_code == 409:
+            conflicts = [(error['extra']['name'], error)
+                         for error in response_json['error_list']
+                         if error['code'] == 'conflict']
+            parts = ["Metadata not updated!"]
+            for field_name, error in sorted(conflicts):
+                sent = metadata.get(field_name)
+                parts.extend((
+                    "Conflict in {!r} field:".format(field_name),
```

Comment:
The error should probably say something like `You can use --force-metadata to update anyway`?

Reply:
And repeat this message in ALL conflict lines, you say? 

Maybe it's better to add a line (in case of conflicts) after all conflicts, saying something like "You can repeat the command with --force-metadata to force the local values into the Store"?

What do you think? Thanks!

Reply:
Yeah, that sounds great!